ui/hooks/useAccountTotalCrossChainFiatBalance.js

```diff
@@ -0,0 +1,102 @@
+import { shallowEqual, useSelector } from 'react-redux';
+import { toChecksumAddress } from 'ethereumjs-util';
+import {
+  getCurrentCurrency,
+  getNetworkConfigurationsByChainId,
+  getCrossChainTokenExchangeRates,
+  getCrossChainMetaMaskCachedBalances,
+} from '../selectors';
+import {
+  getValueFromWeiHex,
+  sumDecimals,
+} from '../../shared/modules/conversion.utils';
+import { getCurrencyRates } from '../ducks/metamask/metamask';
+import { getTokenFiatAmount } from '../helpers/utils/token-util';
+
+export const useAccountTotalCrossChainFiatBalance = (
+  account,
+  formattedTokensWithBalancesPerChain,
+) => {
+  const allNetworks = useSelector(getNetworkConfigurationsByChainId);
+  const currencyRates = useSelector(getCurrencyRates);
+  const currentCurrency = useSelector(getCurrentCurrency);
+
+  const crossChainContractRates = useSelector(
+    getCrossChainTokenExchangeRates,
+    shallowEqual,
+  );
+
+  const crossChainCachedBalances = useSelector(
+    getCrossChainMetaMaskCachedBalances,
+  );
+
+  // const loading = false; //todo check if loading is still needed
```

Comment:
I have this same TODO in my PR. I'm starting to think that it isn't needed anymore.

Reply:
yes was planning to remove this variable

Reply:
We don't really have a way to replicate loading in the same way.   I think there are several reasons it goes away.